plugins/single_plugins/alpha.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ * The MIT License (MIT)
+ * 
+ * Copyright (c) 2018 Scott Moreau
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <core.hpp>
+#include <view.hpp>
+#include <plugin.hpp>
+#include <output.hpp>
+#include <animation.hpp>
+#include "view-transform.hpp"
+#include "workspace-manager.hpp"
+#include <nonstd/make_unique.hpp>
+
+class wayfire_alpha : public wayfire_plugin_t
+{
+    axis_callback axis_cb;
+    wf_duration duration;
+
+    public:
+    void init(wayfire_config *config)
+    {
+        grab_interface->name = "alpha";
+        grab_interface->abilities_mask = WF_ABILITY_CONTROL_WM;
+
+        axis_cb = [=] (wlr_event_pointer_axis* ev)
+        {
+            if (!output->activate_plugin(grab_interface))
+                return;
+
+            output->deactivate_plugin(grab_interface);
+
+            auto focus = core->get_cursor_focus();
+
+            if (!focus)
+                return;
+
+            auto view = core->find_view(focus->get_main_surface());
+
+            auto layer = output->workspace->get_view_layer(view);
+
+            if (layer == WF_LAYER_BACKGROUND)
+                return;
+
+            if (!view->get_transformer("alpha"))
+                view->add_transformer(nonstd::make_unique<wf_2D_view> (view), "alpha");
+
+            if (ev->orientation == WLR_AXIS_ORIENTATION_VERTICAL)
+                update_alpha_target(view, ev->delta);
+        };
+
+        auto section = config->get_section("alpha");
+        auto modifier = section->get_option("modifier", "<alt>");
+
+        output->add_axis(modifier, &axis_cb);
+    }
+
+    void update_alpha_target(wayfire_view view, float delta)
+    {
+        auto transformer = dynamic_cast<wf_2D_view*> (view->get_transformer("alpha").get());
+        float alpha = transformer->alpha;
+
+        alpha -= delta * 0.003;
```

Comment:
maybe add an option for speed?

Reply:
I'm not sure what you mean by this. How would speed value be interpreted?